### DISCUSSION TITLE
Is there a testing framework for connection directive pagination?

### DISCUSSION THREAD

**Discussion by BCantos17** [Netflix/dgs-framework#846] [Q&A] [answered]
I have a schema like this
```
directive @connection on OBJECT

type Query{
    images(first: Int!, after: String): ImageConnection
}

type Image @connection {
    id: String!
}
```

And have a test class in kotlin like this

```
@SpringBootTest(classes = [DgsAutoConfiguration::class])
class ImageFetcherTest {
// test here
...
}
```

But when I run my test, I am getting an error which the message 

> The field type 'ImageConnection' is not present when resolving for type 'Query'

The dgs pagination framework creates the ImageConnection type when spring boot app starts up normally but not for this test. I tried googling and could not find anyone else who encountered this issue. Is there a reference or a guide to test out pagination when using connection directive or is this something not yet supported?

**Comment by BCantos17** [comment, marked as answer]
Managed to get my test to pass, had to add `DgsPaginationTypeDefinitionRegistry` inside the SpringBootTest classes. 

Leaving this open to see if this is the correct way to go about it

**Comment by srinivasankavitha** [comment]
Yes, you do need to specify all the required  classes to load when writing
a test, so that is the correct approach.

**Comment by jandrade1** [comment]
FWIW, I'm including `DgsPaginationAutoConfiguration.class` under SpringBootTest classes instead, which pulls in `DgsPaginationTypeDefinitionRegistry` and I suppose anything else that may be included for pagination in the future.
